src/Voron/Data/VoronStream.cs

```diff
@@ -114,10 +131,13 @@ public override int ReadByte()
                     return -1;
             }
 
-            var page = _llt.GetPage(chunk.PageNumber, _pagerRef);
-            _pagerRef.Pager.EnsureMapped(_llt, _pagerRef.PagerPageNumber, VirtualPagerLegacyExtensions.GetNumberOfOverflowPages(page.OverflowSize));
+            if (!_lastChunk.Page.IsValid || _lastChunk.Page.PageNumber != chunk.PageNumber)
+            {
+                var page = _llt.GetPage(chunk.PageNumber, _pagerRef);
```

Comment:
@redknightlois the usage of `_pageRef` here is very counter intuitive we pass empty object here to get it filled back - would it be any perf issue if we added `ref` here